docs/pages/enroll-resources/database-access/enroll-aws-databases/rds-oracle.mdx

```diff
@@ -0,0 +1,389 @@
+
+## Step 6/9. Start the Database Service
+
+(!docs/pages/includes/start-teleport.mdx service="the Teleport Database Service"!)
+
+## Step 7/9. Create a Teleport users
+
+(!docs/pages/includes/database-access/create-user.mdx!)
+
+## Step 8/9. (Optional) Configure Teleport to pull audit logs from Oracle Audit Trail
+
+Teleport can pull audit logs from Oracle Audit Trail.
+In order to enable this feature, you will need to configure Oracle Audit Trail and create a dedicated Teleport
+user that will be used to fetch audit events from Oracle Audit Trail.
+
+Create an internal Oracle `TELEPORT@EXAMPLE.COM` user that will fetch
+audit events from Oracle Audit Trail:
+
+```sql
+CREATE USER "TELEPORT@EXAMPLE.COM" IDENTIFIED EXTERNALLY;
+GRANT CREATE SESSION TO "TELEPORT@EXAMPLE.COM";
+GRANT SELECT ON SYS.DBA_AUDIT_TRAIL TO "TELEPORT@EXAMPLE.COM";
+GRANT SELECT ON SYS.V_$SESSION TO "TELEPORT@EXAMPLE.COM";
+```
+
+Create the corresponding `TELEPORT@EXAMPLE.COM` user in AD.
+
+Enable standard auditing by modifying the RDS Oracle parameter group `audit_trail` to contain [`db,extended`](https://aws.amazon.com/blogs/database/part-1-security-auditing-in-amazon-rds-for-oracle/#:~:text=Standard%20auditing%20with%20Amazon%20RDS%20for%20Oracle).
+
+Reboot your Oracle instance to propagate audit trail changes.
+
+Enable Oracle auditing for the `alice` user:
+```sql
+AUDIT ALL STATEMENTS by "ALICE@EXAMPLE.COM" BY access;
+```
+
+You must enable auditing for each Teleport user that will be used to connect to Oracle.
+Additionally you can create a different audit policy for each user.
+
+Edit the Database Service configuration to specify the audit user you created earlier.
+
+```diff
+db_service:
+  enabled: "yes"
+  databases:
+  - name: "oracle"
+    protocol: "oracle"
+    uri: "oracle-instance.aabbccddeegg.eu-central-1.rds.amazonaws.com:2484"
+    ad:
+      keytab_file: "/etc/teleport/teleport.keytab"
+      domain: "EXAMPLE.COM"
+      krb5_file: "/etc/krb5.conf"
++   oracle:
++     audit_user: "teleport"
+```
+
+Teleport doesn't clean up audit trail events from Oracle Audit Trail.
+Make sure to configure an Oracle Audit Trail cleanup policy to avoid running out of disk space.
+
+
+## Step 9/9. Connect
+
+Log in to your Teleport cluster. Your Oracle database should appear in the
+list of available databases:
+
+```code
+$ tsh login --proxy=teleport.example.com --user=alice
+$ tsh db ls
+# Name      Description         Labels
+# --------- ------------------- -------
+# oracle                     env=dev
+```
+
+Connect to the database:
+
+```code
+$ tsh db connect --db-user=alice --db-name ORCL oracle
+```
+
+To log out of the database and remove credentials:
+
+```code
+# Remove credentials for a particular database instance.
+$ tsh db logout oracle
+# Remove credentials for all database instances.
+$ tsh db logout
+```
+
+## Next steps
+
+(!docs/pages/includes/database-access/guides-next-steps.mdx!)
+
+## Troubleshooting
```

Comment:
Should we add an item with `Timeout errors` ([similar to what we have in other database guides](https://goteleport.com/docs/enroll-resources/database-access/enroll-aws-databases/rds/#timeout-errors))?

Also, I see we have a lot of errors that can happen due to case sensitive usernames and others (like the realm name). Should we check how those appear to the `tsh` end and have a section about them?

Reply:
Expanded troubleshooting.